pvlib/snow.py

```diff
@@ -143,6 +159,10 @@ def coverage_nrel(snowfall, poa_irradiance, temp_air, surface_tilt,
     snow_coverage.ffill(inplace=True)
     snow_coverage -= cumulative_sliding
 
+    if snow_depth is not None:
+        # no coverage when there's no snow on the ground
+        # described in [2] to avoid non-sliding snow for low-tilt systems.
+        snow_coverage[snow_depth <= 0] = 0.
```

Comment:
I don't think this check can be applied at the end like this.  In cases where the check takes effect (snow_depth = 0 --> coverage = 0), that effect should propagate forward in time and influence values for future timestamps.  As it is now, I don't think this code allows that. 

See for example this input:

```python
times = pd.date_range("2019-01-01", freq="h", periods=4)
snowfall = pd.Series([10, 0, 0, 0.1], index=times)  # last value is below threshold_snowfall
snow_depth = pd.Series([10, 5, 0, 0.1], index=times)
poa_irradiance = pd.Series(100, index=times)
temp_air = pd.Series(-1, index=times)
surface_tilt = 10

coverage_nrel(snowfall, poa_irradiance, temp_air, surface_tilt, snow_depth)

# output:
2019-01-01 00:00:00    1.000000
2019-01-01 01:00:00    0.965791
2019-01-01 02:00:00    0.000000
2019-01-01 03:00:00    0.897374  # this value doesn't make sense, should be zero
```

Reply:
I think you are correct. The [SAM implementation](https://github.com/NREL/ssc/blob/7e7c049e53f073e9a77b7eaa1f64097c3342b725/shared/lib_snowmodel.cpp#L95) appears to iterate over timesteps.

Reply:
In the SAM algorithm, [these lines](https://github.com/NREL/ssc/blob/7e7c049e53f073e9a77b7eaa1f64097c3342b725/shared/lib_snowmodel.cpp#L115C1-L118C1) set coverage to full if both of two conditions are met:

```
	if ((snowDepth - previousDepth) >= deltaThreshold*dt && snowDepth >= depthThreshold){
		coverage = 1;
	}
```

So change in snow depth exceeds deltaThreshold (*dt converts to hourly), and snowDepth exceeds a different threshold so the snow is "sticking" around.